spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/TableWriter.scala

```diff
@@ -151,9 +170,11 @@ class TableWriter[T] private (
 
 object TableWriter {
 
-  private def checkColumns(table: TableDef, columnNames: Seq[String]) = {
+  private def checkColumns(table: TableDef, columnRefs: IndexedSeq[ColumnRef]) = {
```

Comment:
It was probably wrong before, but the order of methods is wrong here. In scala we order methods bottom-up. So references should be always backwards.